frappe/commands/site.py

```diff
@@ -105,16 +106,26 @@ def restore(context, sql_file_path, mariadb_root_username=None, mariadb_root_pas
 			print('Invalid path {0}'.format(sql_file_path[3:]))
 			sys.exit(1)
 
-	if sql_file_path.endswith('sql.gz'):
+	if sql_file_path.endswith('sql.gz') or sql_file_path.endswith('enc.gz'):
 		sql_file_path = extract_sql_gzip(os.path.abspath(sql_file_path))
 
+	if sql_file_path.endswith('.enc') and not decrypt_password:
+		print('Decrypt encrypted sql file using --decrypt-password option.')
+		sys.exit(1)
+
+	if sql_file_path.endswith('.enc') and decrypt_password:
+		output_file_name = sql_file_path.replace(".xb.enc", ".sql")
+		os.system(
+			"openssl aes-256-cbc -d -in " + sql_file_path + " -out " + output_file_name + " -k " + decrypt_password)
+		sql_file_path = output_file_name
+
 	site = get_site(context)
 	frappe.init(site=site)
 	_new_site(frappe.conf.db_name, site, mariadb_root_username=mariadb_root_username,
 		mariadb_root_password=mariadb_root_password, admin_password=admin_password,
 		verbose=context.verbose, install_apps=install_app, source_sql=sql_file_path,
 		force=context.force)
-
+	os.system("rm " + sql_file_path)
```

Comment:
Why are you removing this file?

Reply:
after we decrypt the encrypted file, we need to remove the decrypted SQL file after it is restored.

Reply:
again, we'll replace the os.system into frappe.utils.execute_in_shell. thanks